### PR TITLE
Add hooks for wire editor

### PR DESCRIPTION
**Pull request by Python1320**
I have an use case for these. They could come in handy probably for others too.

### CODE REVIEW

**Comment by CaptainPRICE** [comment]
Add more hooks.

**Comment by TomyLobo** [comment]
Would be nice if you had described your use case instead of just mentioning that you have one :)

**Comment by Python1320** [comment]
I'll see if I can get the hooks to work for things like the in-editor animation.

On metastruct we have a chat bubble over players. If you code Lua in the ingame lua editor the code shows up on the bubble for the curious people or for help, of course you can disable it. I wanted to extend this to E2 and starfall and maybe other gates. This lets me do that. I'm already using it on metastruct if someone wants to test it.

**Comment by TomyLobo** [comment]
I don't want some random server to steal my code, so no.
Same with displaying it to everyone passing by.

**Comment by Nebual** [comment]
I like hooks. I wouldn't allow a default enabled code-broadcasting feature into official wire, but if some server wants to do that, they're welcome to - if you don't like that feature, don't use E2 on that server, or don't visit it.
From a security standpoint, servers can already do this by overriding the functions (like I've done in #1515 to extend Garrydupe), so this doesn't add any new risks.
From a performance standpoint, this runs a hook on _every keypress_, which is rather frequent... I might prefer it in the autosave timer for that reason, but I guess that depends on what code is consuming the hook. However clientside performance isn't as critical, particularly with the editor open, so its not a blocker for me.

File: lua/wire/client/text_editor/wire_expression2_editor.lua [hunk 1]
```diff
@@ -1844,6 +1846,8 @@ function Editor:Close()
 	self.chip = false
 
 	self:SaveEditorSettings()
+	
+        hook.Run("WireEditorClose", self)
```
Comment (CaptainPRICE) [inline lua/wire/client/text_editor/wire_expression2_editor.lua]: What's with these spaces?

**Comment by AbigailBuccaneer** [comment]
bump @Nebual ready to merge?

**Pull request merged**